src/Signup_Command.php

```diff
@@ -0,0 +1,256 @@
+	 * @package wp-cli
+	 */
+	public function list_( $args, $assoc_args ) {
+		global $wpdb;
+
+		if ( isset( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = explode( ',', $assoc_args['fields'] );
+		} else {
+			$assoc_args['fields'] = $this->obj_fields;
+		}
+
+		$signups = array();
+
+		$results = $wpdb->get_results( "SELECT * FROM $wpdb->signups", ARRAY_A );
+
+		if ( $results ) {
+			foreach ( $results as $item ) {
+				// Support features like --active=0.
+				foreach ( array_keys( $item ) as $field ) {
+					if ( isset( $assoc_args[ $field ] ) && $assoc_args[ $field ] !== $item[ $field ] ) {
+						continue 2;
+					}
+				}
+
+				$signups[] = $item;
+			}
+		}
+
+		$format = Utils\get_flag_value( $assoc_args, 'format', 'table' );
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		if ( 'ids' === $format ) {
+			WP_CLI::line( implode( ' ', wp_list_pluck( $signups, 'signup_id' ) ) );
+		} else {
+			$formatter->display_items( $signups );
+		}
+	}
+
+	/**
+	 * Gets details about the signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : Signup ID, user login, user email or activation key.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole signup, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Get a specific subset of the signup's fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Get signup.
+	 *     $ wp signup get 1
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | signup_id | user_login | user_email          | registered          | active | activation_key   |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | 1         | bobuser    | bobuser@example.com | 2024-03-13 05:46:53 | 1      | 663b5af63dd930fd |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *
+	 * @package wp-cli
+	 */
+	public function get( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		$formatter = $this->get_formatter( $assoc_args );
+
+		$formatter->display_items( array( $signup ) );
+	}
+
+	/**
+	 * Activates a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <signup>
+	 * : Signup ID, user login, user email or activation key.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # Activate signup.
+	 *     $ wp signup activate 2
+	 *     Success: Signup activated. Password: bZFSGsfzb9xs
+	 *
+	 * @package wp-cli
+	 */
+	public function activate( $args, $assoc_args ) {
+		$signup = $this->fetcher->get_check( $args[0] );
+
+		if ( $signup ) {
+			$result = wpmu_activate_signup( $signup->activation_key );
+		}
+
+		if ( is_wp_error( $result ) ) {
+			WP_CLI::error( 'Signup could not be activated. Reason: ' . $result->get_error_message() );
```

Comment:
The error message is usually worded differently ([example](https://github.com/wp-cli/entity-command/blob/22d726dd3ce09a7a70ca1ac2c935e2f0450894a4/src/Post_Command.php#L517)).
Also, please add the signup ID. Lastly, does it make sense to only output the reason when the core `--debug` flag is set?
The default output would be this:
```php
			WP_CLI::error( 'Failed activating signup 123.);
```